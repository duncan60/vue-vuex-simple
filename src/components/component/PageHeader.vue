<template>
  <header>
      <h1>Vue with Vuex Simple</h1>
      <nav>
        <ul class="nav-list">
          <router-link class="nav-list__link" to="/todo">Todo</router-link>
          <router-link class="nav-list__link" to="/fetch">Fetch</router-link>
        </ul>
      </nav>
    </header>
</template>

<script>
export default {
  name: 'PageHeader',
};
</script>

<style lang="scss">
.nav-list {
  padding: 0;
  &__link {
    padding: 10px;
  }
}
</style>
