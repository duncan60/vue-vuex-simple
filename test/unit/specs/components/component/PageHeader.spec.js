import { mount } from '@vue/test-utils';
import PageHeader from '@/components/component/PageHeader';

describe('PageHeader.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(PageHeader);
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });
  it('test render h1 title', () => {
    const component = mount(PageHeader);
    const h1 = component.vm.$el.querySelector('h1').textContent;
    expect(h1).toEqual('Vue with Vuex Simple');
  });
});
