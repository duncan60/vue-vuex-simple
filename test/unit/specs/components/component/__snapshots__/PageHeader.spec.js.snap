// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader.vue test initial rendering 1`] = `
<header>
    <h1>Vue with Vuex Simple</h1>
    <nav>
        <ul class="nav-list">
            <router-link to="/todo" class="nav-list__link">Todo</router-link>
            <router-link to="/fetch" class="nav-list__link">Fetch</router-link>
        </ul>
    </nav>
</header>
`;
